### PR TITLE
Add webauthn Banner, Move webauthn above TOTP

### PR DESCRIPTION
**Pull request by ericherscovich**
Contributes to #3800 

## What problem are you solving?
We would like to better publicize webauthn now that it is fully supported. After discussing in the Slack, we have determined that a banner would be a good approach, as well as a flash notification after signin. Further, webauthn is the prefered MFA solution. So, it should be moved higher up in the settings edit page.

## What approach did you choose and why?
Adds a banner to publicize webauthn, links directly to the section to add a device on the settings edit page. Also, moves webauthn content above TOTP in the settings edit page.
Screenshot 2023-07-17 at 2 21 53 PM

When a user signs in using no MFA, or with TOTP, they are also given a flash warning telling them about security devices. Further, the flash warning CSS is getting an update to be sleeker and more readable (after 9 years!). In the image below, you can see the new flash warning.
Screenshot 2023-07-18 at 3 25 54 PM

## What should reviewers focus on?
Content of the banner, styling of the new flash warning. 

## Blockers ⚠️ 
1. This PR should only be merged after the Safari PRs are shipped. https://github.com/rubygems/rubygems.org/pull/3873
https://github.com/rubygems/rubygems/pull/6774
2. This PR should only be merged after the new blog post is live https://github.com/rubygems/rubygems.github.io/pull/164

### CODE REVIEW

File: app/assets/stylesheets/modules/org.css [hunk 1]
```diff
@@ -265,3 +265,20 @@
...
+  font-weight: 500;
+  font-size: 16px;
+  color: white;
+}
+
+#banner a {
+  color: #e9573f;
+  text-decoration: underline;
+  font-weight: bold;
+}
```
Comment (jenshenny) [inline app/assets/stylesheets/modules/org.css:284]:
```suggestion
}

```

File: config/locales/en.yml [hunk 2]
```diff
@@ -147,6 +147,7 @@ en:
       find_blurb: Find, install, and publish RubyGems.
       learn:
         install_rubygems: Install RubyGems
+      mfa_banner_html: 🎉 We now support security devices! Improve your account security by <a href="/settings/edit#security-device">setting up</a> a new device.
```
Comment (jenshenny) [inline config/locales/en.yml:150]: We're planning to also open up a blog post? Should we have both available around the same time? If so, wdyt about a "Learn More" at the end of the message. Eg.

```suggestion
      mfa_banner_html: 🎉 We now support security devices! Improve your account security by <a href="/settings/edit#security-device">setting up</a> a new device. Learn more [here](link to blog post)!
```

**Review by jenshenny — COMMENTED** [review]
Lack of CSS knowledge here to give great advice but I think it'll be beneficial to have some horizontal padding, and a non fixed font-size (using `ems` / `rems`)? Looking at the trend, we do use fixed sizing 🤔 

Screenshot 2023-07-18 at 12 55 56 PM

File: app/assets/stylesheets/modules/org.css [hunk 3]
```diff
@@ -265,3 +265,20 @@
...
+#banner {
+  background-color: #141c22;
+  text-align: center;
+  padding: 12px 0;
+  font-weight: 500;
+  font-size: 16px;
+  color: white;
+}
+
+#banner a {
```
Comment (bettymakes) [inline app/assets/stylesheets/modules/org.css:280]: Change to a class `.banner`

File: app/views/home/index.html.erb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <div class="home__image-wrap">
   <div class="home__image"></div>
 </div>
+<div id="banner"><%=t('.mfa_banner_html')%></div>
```
Comment (bettymakes) [inline app/views/home/index.html.erb:4]: If we plan on using this `id` for other purposes beyond styling (e.g. to link to this page with this element anchored to the top), then we can keep the `id` for _that_ purpose. Otherwise, I'd change the `id` to a `class`.

That said, if we plan on using the `id` for other purposes, let's keep the `id` _and_ add a `class` for styling-specific conerns.

File: app/views/settings/edit.html.erb [hunk 5]
```diff
@@ -13,6 +13,20 @@
     <% end %>
   <% end %>
 
+  <h2 id='security-device'><%= t(".webauthn_credentials") %></h2>
```
Comment (bettymakes) [inline app/views/settings/edit.html.erb:16]: Use double quotes for attribute values:
`<h2 id="security-device">`

File: config/locales/en.yml (hunk 2 above)
Comment (ericherscovich) [reply config/locales/en.yml:150]: That's a great idea. Once the blog post is up and I have a URL to link to I will update the translation with that 👍

**Review by bettymakes — COMMENTED** [review]
I'm torn on the placement of the banner. I think the placement makes more sense above the nav bar. I know our flash messages are typically placed below, but the current placement visually disrupts the flow of the content.

<p align="center">
  <kbd>
    Mock for homepage banner
  </kbd>
</p>

That's the placement I'd propose. This type of placement is not an uncommon pattern (e.g. [1](https://github.com/rubygems/rubygems.org/assets/4270287/380834f6-bb61-49cb-839b-be23d00d3e50), [2](https://github.com/rubygems/rubygems.org/assets/4270287/b7282379-0f8c-4317-b53b-cba9af1dc7e4)).

WDYT?

As for approach, there's two ways we can tackle it:
* Add banner to `application.rb` and only display it if the route matches the homepage path
* Keep the markup where it is (in `index.html.erb`), we just need to add a few extra CSS rules and we'll be able to adjust the placement so the banner moves up to the top.

File: config/locales/en.yml (hunk 2 above)
Comment (bettymakes) [reply config/locales/en.yml:150]: Agreed +1. Let's do that.
A small adjustment - `Learn more` should be the CTA itself.

```
mfa_banner_html: 🎉 We now support security devices! Improve your account security by <a href="/settings/edit#security-device">setting up</a> a new device. [Learn more](link to blog post)!
```

File: app/assets/stylesheets/modules/org.css [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
 .flash {
-  border-bottom: 1px solid #e9573f;
+  border-bottom: 1px solid #e9573f45;
   background-color: white; }
 
 .flash-wrap {
   padding-top: 5px;
   padding-bottom: 5px;
-  background-image: linear-gradient(to right, rgba(233, 87, 63, 0.3), #e9573f); }
+  padding: 8px 0px;
```
Comment (bettymakes) [inline app/assets/stylesheets/modules/org.css:10]: A few notes:
* The `padding-top` & `padding-bottom` rules are redundant. Let's delete them. The `padding` rule on Line 10 overwrites the rules above.
* Can we increase the top/bottom padding by a bit more, increase to `10px`
* For the `0px` left/right padding, drop the `px` unit. Units aren't required when the value is `0`.

```css
.flash-wrap {
  padding: 10px 0;
  background-color: #fff6d2;
}
```

**Review by bettymakes — COMMENTED** [review]
I made a [similar point](https://github.com/rubygems/rubygems.org/pull/3945/files#r1267251590) about not bolding _everything_. Instead, let's only bold CTAs (links) in our flash messages. I'd recommend we drop the `<b>` from the flash message template altogether. Keeping the entire message bolded reduces the impact and emphasis we want to place on CTAs.

Example: Currently all flash messages are bolded:
<p align="center">
  <kbd>
    Flash message completely bolded
  </kbd>
</p>

Example: Proposing we drop the `<b>`:
<p align="center">
  <kbd>
    Flash message, only bolding CTAs
  </kbd>
</p>

**Comment by bettymakes** [comment]
> Lack of CSS knowledge here to give great advice but I think it'll be beneficial to have some horizontal padding, and a non fixed font-size (using `ems` / `rems`)? Looking at the trend, we do use fixed sizing 🤔 cc:@bettymakes


**TL;DR - In this case, either `15px` or `1rem` is fine for now.**

Yea, other than a few rules our team has introduced more recently, `em`s and `rem`s aren't widely used here yet. Ultimately, one of the key appeals of using those relative units (especially `rem`) is so the site can scale accordingly. As you know, if a user adjusts their browser's [font-size](https://github.com/rubygems/rubygems.org/assets/4270287/1d10c92d-5928-4f5d-a6ec-ae0673593bd3) our site would adjust accordingly.

However, our [root font-size is fixed](https://github.com/rubygems/rubygems.org/blob/master/app/assets/stylesheets/base.css#L26), so we wouldn't be able to reap that benefit 😅 . 



> This is also interesting 🤔 Screenshot 2023-07-18 at 2 07 11 PM

lol great catch @jenshenny! Them sharp eyes. It looks like the nav menu has always been transparent. Since the background is a solid colour, we never noticed that until we go and add a banner underneath it 😅 .

File: app/assets/stylesheets/modules/org.css [hunk 7]
```diff
@@ -265,3 +265,20 @@
     float: right; }
   .about__assets__download:focus:before, .about__assets__download:hover:before {
     animation: arrow .75s infinite; }
+
+/* Banner */
+
+#banner {
+  background-color: #141c22;
+  text-align: center;
+  padding: 12px 0;
```
Comment (bettymakes) [inline app/assets/stylesheets/modules/org.css:274]: Per [Jenny's note](https://github.com/rubygems/rubygems.org/pull/3945#issuecomment-1641053570) re: lack of padding, let's add horizontal padding here.

Recommending `15px` all around. Skimming through the spacing-related rules in `home.css`, it looks like they use multiples of 5s (look at the margin and padding rules).

```suggestion
  padding: 15px;
```

File: app/assets/stylesheets/modules/org.css [hunk 8]
```diff
@@ -265,3 +265,20 @@
...
+  font-weight: 500;
+  font-size: 16px;
```
[the remark this answers is not in the document]
Comment (bettymakes) [reply app/assets/stylesheets/modules/org.css:276]: To be thorough, let's also resize the font according to the existing break points on the home page.

I'm observing that there are two main breakpoints:
* Around `900px`
* Around `600px`

Recommending that we bump down the font-size to `12px` when we hit `899px` and smaller. Add a media query that'll look something like this:

```
.banner {
  /* fill in your rules, don't include font-size, will use default which is 15px */
}

@media (max-width: 899px) {
  .banner {
    font-size: 12px; 
  } 
}
```

**Comment by ericherscovich** [comment]
For now, moved the banner above the header using Rails. Will pair with Betty tomorrow to see if this can be done using some fancy CSS.

**Review by bettymakes — COMMENTED** [review]
@ericherscovich & I discussed this on Friday. He passed the torch for me to resolve the remaining issues. We should be good for another round of reviews 👍. 

Eric, feel free to remove the `WIP` label from the PR title. 

---
Updated banner and resizing accordingly per site's breakpoints:
<p align="center">
  <kbd>
    Resizing browser, adjusting font size in banner
  </kbd>
</p>

**Review by jenshenny — APPROVED** [review]
Looks good to me! Would like another review since this introduces some new patterns into the app (banner on the home page, changing up the flash banner style).

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -23,6 +23,9 @@
   </head>
 
   <body class="<%= 'body--index' if request.path_info == '/' %>">
+    <%if request.path == '/'%>
+      <div class="banner"><%=t('.mfa_banner_html')%></div>
```
[the remark this answers is not in the document]
Comment (simi) [reply app/views/layouts/application.html.erb:27]: Comparing the request.path is not good idea (too low-level API). Can't we migrate to current_page? or even use content for and reverse the logic (inject data here from the related template)?

**Comment by simi** [comment]
I did some local testing, looks good!

**Review by simi — APPROVED** [review]

File: config/locales/en.yml [hunk 10]
```diff
@@ -183,6 +183,7 @@ en:
         sign_in: Sign in
         sign_out: Sign out
         sign_up: Sign up
+      mfa_banner_html: 🎉 We now support security devices! Improve your account security by <a href="/settings/edit#security-device">setting up</a> a new device. <a href="#link-to-blog-post">Learn more</a>!
```
[the remark this answers is not in the document]
Comment (bettymakes) [reply config/locales/en.yml]: ✅ Blog post just went live at `https://blog.rubygems.org/2023/08/03/level-up-using-security-devices.html`. PR has been updated with the appropriate URL.
